src/hotspot/share/runtime/arguments.cpp

```diff
@@ -1514,6 +1517,22 @@ jint Arguments::set_ergonomics_flags() {
   return JNI_OK;
 }
 
+void Arguments::set_ergonomics_profile() {
+  if (FLAG_IS_DEFAULT(ErgonomicsProfile)){
+
+#ifdef LINUX
+    if (OSContainer::is_containerized()){
+      FLAG_SET_ERGO(ErgonomicsProfile, "dedicated");
+    }
+#endif //LINUX
+
+  } else {
+    if (strcmp(ErgonomicsProfile, "shared") != 0 && strcmp(ErgonomicsProfile, "dedicated") != 0) {
```

Comment:
could extract to a named function isValidProfile() or something like that.

Reply:
Fixed in 5768a4a9